Samples/syscount: add syscount application

hzy-hits opened the pull request:
Three commits:
1. rex-macros/tracepoint.rs modifications
- add hooks events
2. rex/src/tracepoint/bindings.rs modification 
- add `RawSyscallEnterCtx` and `RawSyscallExitCtx` 
3. Syscount implemtations
- Introduced `loader.c` for argument parsing using static methods.
- Removed typedef declarations for `enum` types throughout the codebase to align with the Linux Kernel coding style.
-  Added the core rex application logic for `syscount`

chinrw commented on samples/syscount/.clang-format:
And could you use `.clang-format` from linux kernel

chinrw replied:
Like 
```
clang-format -i --style=file:../../linux/.clang-format loader.c
```

hzy-hits replied:
Sure, I will remove `.clang-format`

jinghao-jia commented on samples/syscount/loader.c:
```diff
@@ -0,0 +1,448 @@
+
+#define MAX_SYSCALL_ID (sizeof(syscall_names) / sizeof(syscall_names[0]))
+
+static volatile bool exiting = false;
```
`exiting` does not need to be `volatile`.

jinghao-jia commented on samples/syscount/loader.c:
```diff
@@ -0,0 +1,448 @@
+    exiting = true;
+}
+
+typedef enum {
```
`typedef` is still here

jinghao-jia commented on samples/syscount/loader.c:
```diff
@@ -0,0 +1,448 @@
+    char *filter_syscalls;
+};
+
+void print_usage(const char *prog_name) {
```
This function should also be `static`

chinrw commented:
Hi, could you help rewrite the commit message to follow our current style and signoff the commits.

hzy-hits commented:
Three commits:
1. rex-macros/tracepoint.rs modifications
- add hooks events
2. rex/src/tracepoint/bindings.rs modification 
- add `RawSyscallEnterCtx` and `RawSyscallExitCtx` 
3. Syscount implemtations
- Introduced `loader.c` for argument parsing using static methods.
- Removed typedef declarations for `enum` types throughout the codebase to align with the Linux Kernel coding style.
-  Added the core rex application logic for `syscount`

jinghao-jia commented:
The last commit (`Merge branch 'main' into main`) seems unnecessary, can you drop it?

jinghao-jia commented on samples/syscount/loader.c:
Looks like this file is still not formatted?

hzy-hits replied:
updated

jinghao-jia reviewed (approved):
LGTM, thanks for the hard work!

jinghao-jia commented:
@hzy-hits it looks like the [DCO check failure](https://github.com/rex-rs/rex/actions/runs/14655556330?pr=29) is because the author name and the sign-off name are different:

```
Author: Ivena <huangzhy77@outlook.com>
Date:   Thu Apr 24 20:41:47 2025 -0400

    samples/syscount: add syscount tool

    Signed-off-by: Zhenyu Huang <huangzhy77@outlook.com>
```

hzy-hits commented:
> @hzy-hits it looks like the [DCO check failure](https://github.com/rex-rs/rex/actions/runs/14655556330?pr=29) is because the author name and the sign-off name are different (both the first and the third commit have this problem):
> 
> ```
> Author: Ivena <huangzhy77@outlook.com>
> Date:   Thu Apr 24 20:41:47 2025 -0400
> 
>     samples/syscount: add syscount tool
> 
>     Signed-off-by: Zhenyu Huang <huangzhy77@outlook.com>
> ```
> 
> Are you directly using the `Signed-off-by` tag generated by git?

I rebased the commits and signed with author's name. Now all commits' sign-off are generated directly. Hope this time it would pass the workflow check.

Thanks for your patience and guidance. I learn quiet a lot from this PR

jinghao-jia commented:
> > @hzy-hits it looks like the [DCO check failure](https://github.com/rex-rs/rex/actions/runs/14655556330?pr=29) is because the author name and the sign-off name are different (both the first and the third commit have this problem):
> > ```
> > Author: Ivena <huangzhy77@outlook.com>
> > Date:   Thu Apr 24 20:41:47 2025 -0400
> > 
> >     samples/syscount: add syscount tool
> > 
> >     Signed-off-by: Zhenyu Huang <huangzhy77@outlook.com>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Are you directly using the `Signed-off-by` tag generated by git?
> 
> I rebased the commits and signed with author's name. Now all commits' sign-off are generated directly. Hope this time it would pass the workflow check.
> 
> Thanks for your patience and guidance. I learn quiet a lot from this PR

No problem! Sorry that we requested so many changes ;)
